engine/apps/google/tasks.py

```diff
@@ -118,5 +128,11 @@ def sync_out_of_office_calendar_events_for_user(google_oauth2_user_pk: int) -> N
 
 @shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True)
 def sync_out_of_office_calendar_events_for_all_users() -> None:
-    for google_oauth2_user in GoogleOAuth2User.objects.filter(user__organization__deleted_at__isnull=True):
+    # some existing tokens may not have all the required scopes, lets skip these
+    tokens_containing_required_scopes = GoogleOAuth2User.objects.filter(
+        *[Q(oauth_scope__contains=scope) for scope in constants.REQUIRED_OAUTH_SCOPES],
+        user__organization__deleted_at__isnull=True,
+    )
+
```

Comment:
Would it be useful to have a `x/total` log message here so we know how we are doing for users correcting not having the required scopes?

Reply:
Good idea 👍 added in https://github.com/grafana/oncall/pull/4802/commits/6ec1cb13589616bdcb42ee85db864f7bb6f81186